src/modules/cpu/kernel/fog.hpp

```diff
@@ -0,0 +1,1220 @@
+                                Rpp32f *fogIntensityMask,
+                                Rpp32f *gammaTensor,
+                                RpptROIPtr roiTensorPtrSrc,
+                                RpptRoiType roiType,
+                                RppLayoutParams layoutParams,
+                                rpp::Handle& handle)
+{   
+    Rpp32u numThreads = handle.GetNumThreads();
+    RpptROI roiDefault = {0, 0, (Rpp32s)srcDescPtr->w, (Rpp32s)srcDescPtr->h};
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for(int batchCount = 0; batchCount < dstDescPtr->n; batchCount++)
+    {
+        RpptROI roi;
+        RpptROIPtr roiPtrInput = &roiTensorPtrSrc[batchCount];
+        compute_roi_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+        Rpp32f gammaFactor = gammaTensor[batchCount];
+
+        Rpp8u *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+
+        Rpp8u *srcPtrChannel, *dstPtrChannel;
+        srcPtrChannel = srcPtrImage + (roi.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi.xywhROI.xy.x * layoutParams.bufferMultiplier);
+        dstPtrChannel = dstPtrImage;
+
+        std::random_device rd;  // Random number engine seed
+        std::mt19937 gen(rd()); // Seeding rd() to fast mersenne twister engine
+        std::uniform_int_distribution<> distribX(0, srcDescPtr->w - roi.xywhROI.roiWidth);
+        std::uniform_int_distribution<> distribY(0, srcDescPtr->h - roi.xywhROI.roiHeight);
+
+        RppiPoint maskLoc;
+        maskLoc.x = distribX(gen);
+        maskLoc.y = distribY(gen);
+
+        Rpp32f *fogAlphaMaskPtr, *fogIntensityMaskPtr;
+        fogAlphaMaskPtr = fogAlphaMask + ((srcDescPtr->w * maskLoc.y) + maskLoc.x);
+        fogIntensityMaskPtr = fogIntensityMask + ((srcDescPtr->w * maskLoc.y) + maskLoc.x);
+        
+        Rpp32u bufferLength = roi.xywhROI.roiWidth * layoutParams.bufferMultiplier;
+        Rpp32u alignedLength = (bufferLength / 48) * 48;
+        Rpp32u vectorIncrement = 48;
+        Rpp32u vectorIncrementPerChannel = 16;
+
+#if __AVX2__
+        __m256 pGamma = _mm256_set1_ps(gammaFactor);
+#endif
+        // Fog without fused output-layout toggle (NHWC -> NCHW)
+        if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NHWC) && (dstDescPtr->layout == RpptLayout::NCHW))
+        {
+            Rpp8u *srcPtrRow, *dstPtrRowR, *dstPtrRowG, *dstPtrRowB;
+            Rpp32f *fogAlphaMaskPtrRow, *fogIntensityMaskPtrRow;
+            srcPtrRow = srcPtrChannel;
+            dstPtrRowR = dstPtrChannel;
+            dstPtrRowG = dstPtrRowR + dstDescPtr->strides.cStride;
+            dstPtrRowB = dstPtrRowG + dstDescPtr->strides.cStride;
+            fogAlphaMaskPtrRow = fogAlphaMaskPtr;
+            fogIntensityMaskPtrRow = fogIntensityMaskPtr;
+            for(int i = 0; i < roi.xywhROI.roiHeight; i++)
```

Comment:
Please add space between for and (
```suggestion
            for (int i = 0; i < roi.xywhROI.roiHeight; i++)
```
Applicable to all such instances in the PR